### ISSUE TITLE
Swift 3 Support

### ISSUE THREAD

**Issue by zhigang1992** [SwiftyJSON/SwiftyJSON#647]
First of all, apologize for this really really late response.
### tl;dr

We have landed Swift3 support in master branch, also Swift2.3 support in [swift2](https://github.com/SwiftyJSON/SwiftyJSON/tree/swift2) branch.

<hr>
#### With CocoaPods

Update your Podfile

```
pod 'SwiftyJSON', '3.0.0'
```

Also, at the time of writing, CocoaPods 1.1 is still not released.
So in order to get it working with Xcode8, we need the rc version

```
gem install cocoapods -v 1.1.0.rc.2
```

Or, if you can't upgrade your cocoapods, try this solution from [RxSwift](https://github.com/ReactiveX/RxSwift#cocoapods)

``` ruby
post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['SWIFT_VERSION'] = '3.0'
      config.build_settings['MACOSX_DEPLOYMENT_TARGET'] = '10.10'
    end
  end
end
```
#### Carthage

`github "SwiftyJSON/SwiftyJSON" "3.0.0"`
#### CocoaPods

Update your Podfile

```
pod 'SwiftyJSON', '2.4.0'
```
#### Carthage

`github "SwiftyJSON/SwiftyJSON" "swift2"`

Please file new issues or comment if you're still having issues upgrading

<hr>

Unfortunately, all the previous notification and mentioning to the authors had somehow fall through the cracks.

Even through Swift3 support was something hanging in our mind, but we all thought that the other guy would had done it already.

like a old Chinese saying 

> 三个和尚没水喝 / too many cooks spoil the broth

None of us actually go ahead and get this through.

We are very sorry for this.

Starting from now, I'll be more involved in this project, responding to issues, PRs and pushing out new roadmaps.

<hr>

Thank you all for being involved here, especially for sending out those swift3 swift2.3 PRs.
We look through most of them, and most of them were alike, so we picked the [most recent one](https://github.com/SwiftyJSON/SwiftyJSON/pull/633) from [jeneumeier](https://github.com/jeneumeier).

Cheers

**Comment by kas-kad** [comment]
@zhigang1992 
why don't you set this in your podspec?
```
s.pod_target_xcconfig =  {
        'SWIFT_VERSION' => '3.0',
}```
I can't make the library compile on Xcode 8 after each `pod update` unless I manually set `Use Legacy Swift Language Version` to `No`

**Comment by oliverfoggin** [comment]
I try to use this through Cocoapods but it asks me to convert to Swift 3 because it is written in an old version of Swift.

How can I get the Swift 3.0 version?

I'm using Cocoapod version 3.1.0

**Comment by oliverfoggin** [comment]
@kas-kad I've worked around this for now by forking the repo and specifying my own repo in the Podfile.

Hope that helps for now.

**Comment by kas-kad** [comment]
@oliverfoggin yeah, I mean there are several ways to workaround the problem, but it can be solved once and forever.

**Comment by zhigang1992** [comment]
@kas-kad I think it's a really good idea. Let's add it. 🍻

**Comment by zhigang1992** [comment]
Seems like @fishcharlie already have a PR for it.

https://github.com/SwiftyJSON/SwiftyJSON/pull/673

merging away.

**Comment by gadget00** [comment]
not working for me; Xcode still tells me SwiftyJSON is not compatible with Swift3

**Comment by oliverfoggin** [comment]
@gadget00 you need to install the prerelease version of cocoa pods.

Suso gem install cocoapods --pre

**Comment by DavidLari** [comment]
Actually, release version 1.1 should do it. Pre is on 1.2 beta now.

**Comment by liufsd** [comment]
swift 3.0.1 xcode 8.1

**Comment by delajose90** [comment]
Updating pod 'SwiftyJSON', '3.0.0' is working for me. Really thanks!
